### PR TITLE
Better expansion for nested menus

### PR DESCRIPTION
**Pull request by petrklus**
The current o-hierarchical-hav and its horizontal theme did not allow arbitrary nested menus. This pull request add this support:

Demo: http://cnkpreview.s3.amazonaws.com/ft1_origami/o-hierarchical-nav/better_expansion/responsive-nav-cloneable.html

### CODE REVIEW

File: src/scss/theme-horizontal.scss [hunk 1]
```diff
@@ -79,6 +91,8 @@
 		background-color: oColorsGetPaletteColor('grey-tint4');
 	}
 
+
```
Comment (AlbertoElias) [inline src/scss/theme-horizontal.scss]: just one new line is enough

File: src/scss/theme-horizontal.scss [hunk 2]
```diff
@@ -54,6 +54,18 @@
 		left: 100%;
 		top: 0;
 	}
+	
+	// level X, anything above 3
+	.o-hierarchical-nav__parent.o-hierarchical-nav__outside-right > [data-o-hierarchical-nav-level] {
+		position: absolute;
+		left: 100%;
+		top: 0;
+	}
```
Comment (AlbertoElias) [inline src/scss/theme-horizontal.scss]: new line under here

**Comment by AlbertoElias** [comment]
What's the use case for having so many levels of navigation? Where are you using o-hierarchical-nav. If you need it, it'll be fine to merge as this module will probably be deprecated slowly with the new o-header release

**Comment by petrklus** [comment]
Should be fixed @AlbertoElias. Our use case is to put this menu into a CMS-editable site we are building for the B2B team. While we will most likely not use that as many menu levels as can be seen in the demo, we need to give our users  the flexibility to use the CMS to its full potential (and have its features reflected on the front end).

On another note, I am sad to hear that there are plans for deprecation of this - in our perspective, there are often use-cases when we do not want to use the full blown o-header. Having a small and flexible menu system that can be slotted into any other layouts is quite important from my perspective.

**Comment by AlbertoElias** [comment]
Ok, we'll keep that in mind when discussing the deprecation of the module and we'll keep you in the loop. I'll merge this

**Pull request merged**